openupgrade_scripts/scripts/purchase/17.0.1.2/pre-migration.py

```diff
@@ -0,0 +1,10 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version=None):
+    if not openupgrade.column_exists(env.cr, "purchase_order_line", "discount"):
+        openupgrade.logged_query(
+            env.cr, "ALTER TABLE purchase_order_line ADD COLUMN discount FLOAT"
+        )
+        openupgrade.logged_query(env.cr, "UPDATE purchase_order_line SET discount=0")
```

Comment:
not sure if it's better this way or if do "ADD COLUMN discount FLOAT set default 0" and later remove the default instead.

Reply:
I think we'd need a *lot* of record for it to make much difference, but then it doesn't hurt either